### PR TITLE
Code format

### PR DESCRIPTION
**Pull request by MyrikLD**
1. Black code format https://github.com/psf/black
2. Add badges to the index page (black, version, downloads)
3. Simplified few ifs (without changes in logic)
4. Fix "not x in []"
5. Sort imports
6. Add .idea folder to .gitignore

### CODE REVIEW

File: lib/logitech_receiver/status.py [hunk 1]
```diff
@@ -58,266 +59,249 @@
...
-		count = len(self._receiver)
-		return (_("No paired devices.") if count == 0 else
-			ngettext("%(count)s paired device.", "%(count)s paired devices.", count) % { 'count': count })
-	__unicode__ = __str__
-
-	def changed(self, alert=ALERT.NOTIFICATION, reason=None):
-		# self.updated = _timestamp()
-		self._changed_callback(self._receiver, alert=alert, reason=reason)
-
-	# def poll(self, timestamp):
```
Comment (MyrikLD) [inline lib/logitech_receiver/status.py]: Removed commented code

File: lib/logitech_receiver/status.py [hunk 2]
```diff
@@ -58,266 +59,249 @@
...
-			# (meaning just when the program started or a new receiver was just
-			# detected), pop-up a notification about it
-			alert |= ALERT.NOTIFICATION
-		self.updated = timestamp
-
-		# if _log.isEnabledFor(_DEBUG):
-		# 	_log.debug("device %d changed: active=%s %s", d.number, self._active, dict(self))
-		self._changed_callback(d, alert, reason)
-
-	# def poll(self, timestamp):
```
Comment (MyrikLD) [inline lib/logitech_receiver/status.py]: Removed commented code

**Comment by FFY00** [comment]
> Black code format psf/black

I am not sure about black. I am not really a fan of some of their choices.

Example:
```
kind = (_DK.mouse if 'Mouse' in name
		else _DK.keyboard if 'Keyboard' in name
		else _DK.numpad if 'Number Pad' in name
		else _DK.touchpad if 'Touchpad' in name
		else _DK.trackball if 'Trackball' in name
		else None)
```
turns to
```
kind = (
	_DK.mouse
	if "Mouse" in name
	else _DK.keyboard
	if "Keyboard" in name
	else _DK.numpad
	if "Number Pad" in name
	else _DK.touchpad
	if "Touchpad" in name
	else _DK.trackball
	if "Trackball" in name
	else None
)
```

Which is significantly less readable.

I really which you would have contacted us first before doing all this work.

Either way, we can see what @Lekensteyn thinks.

> Add badges to the index page (black, version, downloads)
> Simplified few ifs (without changes in logic)

> Sort imports

That is fine :)

> Fix "not x in []"

Can you elaborate?

> Add .idea folder to .gitignore

I think this should be in your global .gitignore instead (`~/.gitignore`)

**Comment by MyrikLD** [comment]
It didn’t take much time. Most were done automatically.
Yes, I also drew attention to this construction. It seemed unreadable to me initially. I suggest something like this solution.

```
class _DK(int, Enum):
    mouse = 1
    keyboard = 2
    numpad = 3
    touchpad = 4
    trackball = 5

    @classmethod
    def from_name(cls, name: str):
        for k, v in cls._mapping().items():
            if k in name:
                return v

    @classmethod
    def _mapping(cls):
        return {
            "Mouse": cls.mouse,
            "Keyboard": cls.keyboard,
            "Number Pad": cls.numpad,
            "Touchpad": cls.touchpad,
            "Trackball": cls.trackball,
        }
```

> Fix "not x in []"
```
if not 'cmd' in args:
if not featureId in device.features:
```

> Add .idea folder to .gitignore
Maybe. Didn't think about it. 
I will remove it.

**Comment by FFY00** [comment]
> It didn’t take much time. Most were done automatically.
> Yes, I also drew attention to this construction. It seemed unreadable to me initially. I suggest something like this solution.

That solution is better, but this was just an example, I can give others.

Ignoring my personal dislike for black, I am reluctant to enforce a style considering the current status of the project. Right now we just want to make things work, this project is basically on life support. I think we should revisit it in the future when we have a good development environment in place (CI and more modern code). Does that make sense to you?

> if not 'cmd' in args:
> if not featureId in device.features:

I see, thanks!

> Maybe. Didn't think about it.
> I will remove it.

No problem, we should actually cleanup the current .gitignore.

**Comment by MyrikLD** [comment]
I'm too lazy to format the code manually :)
In any case, you have to bring everything to some kind of format,
and black guarantees that everyone will have the same format. flake8 does not give such guarantees, so sometimes in the review process, you have to read line breaks, quotation marks, etc.

**Comment by FFY00** [comment]
Okay, @Lekensteyn?

**Comment by Lekensteyn** [comment]
Usually I use `autopep8` to ensure a consistent coding style, but I don't mind if you use another formatter. What command did you use to enforce the coding style? Could you add this to the commit message?

If we do proceed with the reformatting, then we could also add CI integration (such as `.travis.yml`) to enforce the coding style for PRs. Something like `the_formatting_command && git diff --exit-code` (which will exit with 1 if there was a difference).

**Comment by MyrikLD** [comment]
Yes, I already have githook for this case

```
#!/bin/sh

FILES=$(git diff --cached --name-status | awk '$1 == "M" || $1 == "A" { print $2}' | grep '\.py$')

if black --check ${FILES}
then
    echo 'Black test passed'
    exit 0
else
    echo 'Commit failed. You forgot to run "black ./". Running BLACK now:'
    black ${FILES}
    echo 'Done. Add reformatted files to your commit and try again'
    exit 1
fi
```
also, you can run black for whole project: `black ./`

**Comment by FFY00** [comment]
Yes, and that's what I meant. If we setup a code style we should enforce it, otherwise it doesn't make sense.

> I am reluctant to enforce a style considering the current status of the project. Right now we just want to make things work, this project is basically on life support.

I don't think we want to stall contributions because they don't follow the code style.

I don't want to hurt anyone's feelings but I keep my position. I don't think it makes much sense to enforce a code style just for now.

However, I leave it open to @Lekensteyn to make the final call.

**Comment by FFY00** [comment]
Closing this for now.

**Pull request closed without merging**